conda/cli/install.py

```diff
@@ -361,123 +408,122 @@ def install(args, parser, command="install"):
-                        platform=None,
-                        use_local=index_args["use_local"],  # --use-local
-                        # use_cache=index_args["use_cache"],  # --use-index-cache
-                        # unknown=index_args["unknown"],  # --unknown
-                        prefix=prefix,
-                        repodata_fn=repodata_fn,
-                    )
-                revision_idx = get_revision(args.revision)
-                with get_spinner(f"Reverting to revision {revision_idx}"):
-                    unlink_link_transaction = revert_actions(
-                        prefix, revision_idx, index
-                    )
-            else:
-                solver_backend = context.plugin_manager.get_cached_solver_backend()
-                solver = solver_backend(
-                    prefix,
-                    context_channels,
-                    context.subdirs,
-                    specs_to_add=specs,
-                    repodata_fn=repodata_fn,
-                    command=args.cmd,
-                )
-                update_modifier = context.update_modifier
-                if (isinstall or isremove) and args.update_modifier == NULL:
-                    update_modifier = UpdateModifier.FREEZE_INSTALLED
-                deps_modifier = context.deps_modifier
-                if isupdate:
-                    deps_modifier = context.deps_modifier or DepsModifier.UPDATE_SPECS
-
+    update_modifier = context.update_modifier
+    if (isinstall or isremove) and args.update_modifier == NULL:
+        update_modifier = UpdateModifier.FREEZE_INSTALLED
+    deps_modifier = context.deps_modifier
+    if isupdate:
+        deps_modifier = context.deps_modifier or DepsModifier.UPDATE_SPECS
+
+    for repodata_fn in Repodatas(
+        repodata_fns,
+        index_args,
+        (UnsatisfiableError, SpecsConfigurationConflictError, SystemExit),
+    ):
+        with repodata_fn as repodata:
+            solver_backend = context.plugin_manager.get_cached_solver_backend()
+            solver = solver_backend(
+                prefix,
+                context_channels,
+                context.subdirs,
+                specs_to_add=specs,
+                repodata_fn=repodata,
+                command=args.cmd,
+            )
+            try:
                 unlink_link_transaction = solver.solve_for_transaction(
                     deps_modifier=deps_modifier,
                     update_modifier=update_modifier,
                     force_reinstall=context.force_reinstall or context.force,
                     should_retry_solve=(
-                        _should_retry_unfrozen or repodata_fn != repodata_fns[-1]
+                        _should_retry_unfrozen or repodata != repodata_fns[-1]
                     ),
                 )
-            # we only need one of these to work.  If we haven't raised an exception,
-            #   we're good.
-            break
```

Comment:
We are not breaking early if successful anymore?

Reply:
We are. This part of the control flow is handled in the iterator/context manager bit. See https://github.com/conda/conda/pull/14670/files#diff-30852b2709103c10b41b1639d623f9d7eb8588f44f24ccf8cb35e8b4171866b0R238 and https://github.com/conda/conda/pull/14670/files#diff-30852b2709103c10b41b1639d623f9d7eb8588f44f24ccf8cb35e8b4171866b0R187